packages/edit-site/src/components/sidebar-navigation-screen-navigation-menu/rename-modal.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+	TextControl,
+	Modal,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+
+export default function RenameModal( { menuTitle, onClose, onSave } ) {
+	const [ editedMenuTitle, setEditedMenuTitle ] = useState( menuTitle );
```

Comment:
We'll need to `decodeEntities` here and also ensure the button disabled check accounts for this.